docs/integrations/sources/notion-migrations.md

```diff
@@ -1,5 +1,24 @@
 # Notion Migration Guide
 
+## Upgrading to 3.0.0
+
+We're continuously striving to enhance the quality and reliability of our connectors at Airbyte. As part of our commitment to delivering exceptional service, we are transitioning source Notion from the Python Connector Development Kit (CDK) to our innovative low-code framework. This is part of a strategic move to streamline many processes across connectors, bolstering maintainability and freeing us to focus more of our efforts on improving the performance and features of our evolving platform and growing catalog. However, due to differences between the Python and low-code CDKs, this migration constitutes a breaking change for users syncing data from the `Comments` stream.
+
+Specifically, we’ve evolved and standardized how state is managed for incremental streams that are nested within a parent stream. This change impacts how individual states are tracked and stored for each partition, using a more structured approach to ensure the most granular and flexible state management. To gracefully handle these changes for your existing connections, we highly recommend resetting your data for the `Comments` stream before resuming your syncs with the new version.
```

Comment:
Is the reset optional? Wondering whether we should be prepared for issues someone declines to do the reset.

Reply:
Yes, resetting the streams is optional, and I'm pretty sure we do encounter issues with people's connections breaking due to them not going through the proper migration steps. I took a look on Metabase when I was first working on the migration, and the rate of adoption for this stream is extremely low (it was < 3% of active connections when I looked). I ran some tests, and thanks to the per-stream error handling Brian Lai has been working on, even if no action is taken, the rest of the streams will sync successfully regardless; because of this my gut says the potential impact is pretty low? Happy to put more thought into addressing this though! Do you think adding some context in the migration guide on the potential consequences of not updating would suffice?